gpt_index/composability/joint_qa_summary.py

```diff
@@ -55,8 +56,13 @@ def build_graph_from_documents(
         documents: Sequence[Document],
     ) -> "ComposableGraph":
         """Build graph from index."""
-
+        event_id = self._service_context.callback_manager.on_event_start(
```

Comment:
curious: how come we don't just log this here instead of at the node parser level ?

Reply:
It's a little more complicated to log at the node parser level. Then the callback manager + handler(s) need to be passed into the node parser, embeddings, llm predictor, etc.

My initial idea was to add the callback manager to each of those objects, and add hook everything up when `ServiceContext.from_defaults` is called, but that seemed a little messy. But I can switch back to that if that makes more sense.

Reply:
I started to switch this around, but tbh I like it better where it is now. Yes, it's a little annoying to have to put the hooks everywhere. But, it retains the useful outside context of what's going on in llama index.

If it's inside the llm_predictor or embedding model, the scope is quite narrow 🤔

Reply:
sounds good, let's keep it as is then.